core/sql/sqlcomp/CmpSeabaseDDLview.cpp

```diff
@@ -1498,32 +1577,50 @@ PrivColumnBitmap colGrantableBitmap;
       }
       int32_t usedColNumber = naCol->getPosition();
      
-      // Grab privileges from the NATable structure
-      PrivMgrUserPrivs *privs = naTable->getPrivInfo();
-      if (privs == NULL) 
+      PrivMgrUserPrivs privs;
+      PrivMgrUserPrivs *pPrivInfo = NULL;
+      if (viewCreator)
+        pPrivInfo = naTable->getPrivInfo();
+      else
+      {
+        PrivStatus retcode = privInterface.getPrivileges((int64_t)naTable->objectUid().get_value(),
+                                                           naTable->getObjectType(),
+                                                           naTable->getOwner(),
+                                                           privs);
+
+        if (retcode == STATUS_ERROR)
+        {
+           *CmpCommon::diags() << DgSqlCode(-CAT_UNABLE_TO_RETRIEVE_PRIVS);
+           return false;
+        }
+        pPrivInfo = &privs;
```

Comment:
I see... so in the one code path I alluded to, pPrivInfo is not pointing to a heap-allocated object but is pointing to a member of an object (possibly heap-allocated), and other code will take care of destroying that larger object.